### ISSUE TITLE
KeySet error if input is not an array

### ISSUE THREAD

**Issue by pengu-fr** [Respect/Validation#670]
Using KeySet rule to validate anything else than an array with raise an error :

`v::keySet(v::key('test'))->assert(null);`
> array_key_exists() expects parameter 2 to be array, null given

I think hasValidStructure function should simply return false if input is not an array :

```
private function hasValidStructure($input)
    {
        if (!is_array($input)){
            return false;
        }
        foreach ($this->getRules() as $keyRule) {
            if (!array_key_exists($keyRule->reference, $input) && $keyRule->mandatory) {
                return false;
            }

            unset($input[$keyRule->reference]);
        }

        return (count($input) == 0);
    }
```

**Comment by batusa** [comment]
Hi @pengu-fr ,

thank you for reporting this issue! a patch is on the way!

**Comment by henriquemoody** [comment]
Fixed on versions 1.0.6 and 1.1.1. 

Thanks for reporting! 🐼
